### ISSUE TITLE
Couldn't attach file in Gmail

### ISSUE THREAD

**Issue by ZacVND** [katzer/cordova-plugin-email-composer#198]
Hi,

I have an Ionic application using this plugin which will send an email to a recipient with a report.pdf file attached.
Here is the code:

```
     if(ionic.Platform.isIOS()) {
       pathFile = cordova.file.documentsDirectory;
     }
     else {
       pathFile = cordova.file.externalRootDirectory;
     }

    $cordovaFile.checkFile(pathFile, "report.pdf")
     .then(function (success) {
        attach = ['' + pathFile.replace('file://','') + "report.pdf"];
      }, function (error) {
        attach = [];
      });

    $cordovaEmailComposer.isAvailable().then(function() {
       // is available
       var toC = $localstorage.getObject('clinician');
        var email = {
          to: toC.email,
          attachments: attach,
          subject: 'My Self Monitoring Sheet',
          body: bodyText,
          isHtml: true
        };

      $cordovaEmailComposer.open(email).then(null, function () {
         // user cancelled email
      });
     }, function () {
       // not available
     });
```

The code works correctly on iOS 9.3 and HTC One M8.
The issue is that on HTC One M8 (and Android in general), it only work correctly with the Mail app but on Gmail app, it will open the email draft with the correct body but will give error "couldn't attach file" so there is no attachment. I'd like some help. Thanks.

The report.pdf file does get created, this is the issue with Gmail app

**Comment by onesmallfry** [comment]
I think your problem is here: `attach = ['' + pathFile.replace('file://','') + "report.pdf"];`.  Removing `file://` is necessary in iOS, but Android (Gmail) needs it.  I have tested this in Android 6.0.1 with Gmail and Outlook using cordova-plugin-email-composer@0.8.2 (0.8.3 does not work in any Android tests I've run).

**Comment by JuanjoSalvador** [comment]
Can't attach any file in Gmail app (compiled using API 23, tested in Android 4.4) with `file://` in string. Gmail app doesn't show any errors, but there isn't any image attached on the email to sent.

**Comment by onesmallfry** [comment]
There is an issue with 0.8.3 and Android for many of us as can be seen in other threads here and elsewhere.  The only workaround that I've found is to use 0.8.2 for Android (however I'm having trouble with it working with iOS 9).  Though, I haven't yet been successful in getting both versions to coexist simultaneously (specifying plugin version of same plugin per platform).

**Comment by hgehlhausen** [comment]
Ensure that you have added the preference
`<preference name="AndroidPersistentFileLocation" value="Compatibility" />`

**Comment by JuanjoSalvador** [comment]
Hey @hgehlhausen, the @ankur-p's workaround works fine. This could be fixed on 0.8.4

**Comment by katzer** [comment]
This a good hint. Since the plugin can be used without the need to such attachments, I won't set the preference from plugin side. That should be done carefully on app side.
